### PR TITLE
(QENG-1781) updated unix path entries for AIO

### PR DESCRIPTION
**Pull request by kevpl**

### CODE REVIEW

**Comment by kevpl** [comment]
hold off on this one.  There are apparently more changes, and I'll need to circle up with people to get those confirmed

**Comment by kevpl** [comment]
failures unrelated to changes

File: lib/beaker/host/unix.rb [hunk 1]
```diff
@@ -58,12 +58,14 @@ def self.foss_defaults
...
-      h['puppetvardir'] = '/opt/puppetlabs/agent/cache'
-      h['distmoduledir'] = '/opt/puppetlabs/agent/modules'
-      h['sitemoduledir'] = '/etc/puppetlabs/agent/modules'
-      h['hieraconf'] = ' /etc/puppetlabs/agent/hiera.conf'
+      h['puppetbindir']   = '/opt/puppetlabs/agent/bin'
+      h['puppetbin']      = "#{h['puppetbindir']}/puppet"
+      h['puppetpath']     = '/etc/puppetlabs/agent'
+      h['puppetvardir']   = '/opt/puppetlabs/agent/cache'
+      h['distmoduledir']  = '/opt/puppetlabs/agent/modules'
+      h['sitemoduledir']  = "#{h['puppetpath']}/code/modules"
```
Comment (nwolfe) [inline lib/beaker/host/unix.rb]: I bet we'll kick ourselves in the near future if we don't put in the new $codedir.

What do you think about adding `h['puppetcodedir'] = "#{h['puppetpath]}/code"` and then rooting these at `puppetcodedir`?

**Comment by nwolfe** [comment]
Shoot, I just realized that we might need to make these changes to the FOSS/PE defaults too, not just AIO.

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
a change was just needed, so I killed the test.  Restarting now

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@anodelman failures unrelated to change

File: lib/beaker/host/unix.rb [hunk 2]
```diff
@@ -58,12 +62,16 @@ def self.foss_defaults
...
-      h['puppetpath'] = '/etc/puppetlabs/agent'
-      h['puppetvardir'] = '/opt/puppetlabs/agent/cache'
-      h['distmoduledir'] = '/opt/puppetlabs/agent/modules'
-      h['sitemoduledir'] = '/etc/puppetlabs/agent/modules'
-      h['hieraconf'] = ' /etc/puppetlabs/agent/hiera.conf'
+      h['puppetbindir']   = '/opt/puppetlabs/agent/bin'
+      h['puppetbin']      = "#{h['puppetbindir']}/puppet"
+      h['puppetpath']     = '/etc/puppetlabs/agent'
+      h['puppetconfdir']  = "#{h['puppetpath']}/config"
+      h['puppetcodedir']  = "#{h['puppetpath']}/code"
```
Comment (joshcooper) [inline lib/beaker/host/unix.rb]: We need to add this property to the windows host too, it should have the same value as `puppetconfdir`

File: lib/beaker/host/unix.rb [hunk 3]
```diff
@@ -42,9 +44,11 @@ def self.foss_defaults
         'puppetserver-confdir' => '/etc/puppetserver/conf.d',
         'puppetservice'     => 'puppetmaster',
         'puppetpath'        => '/etc/puppet',
+        'puppetconfdir'     => '/etc/puppet',
         'puppetvardir'      => '/var/lib/puppet',
         'puppetbin'         => '/usr/bin/puppet',
         'puppetbindir'      => '/usr/bin',
+        'systembindir'      => '/usr/bin',
```
Comment (joshcooper) [inline lib/beaker/host/unix.rb]: does this need to be in AIO also? If so does this need to be in the `windows` host too?

File: lib/beaker/host/unix.rb (hunk 2 above)
Comment (kevpl) [inline lib/beaker/host/unix.rb]: adding the Windows values will be done in QENG-1721

File: lib/beaker/host/unix.rb (hunk 3 above)
Comment (kevpl) [inline lib/beaker/host/unix.rb]: The `aio_defaults` are an override of `foss_defaults`, so it gets that value for free

**Comment by anodelman** [comment]
Windows pathing updates will be made in a separate PR.  Merging this.

**Pull request merged**